src/MetaMorpho.sol

```diff
@@ -201,8 +202,11 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         if (newTimelock > timelock) {
             _setTimelock(newTimelock);
         } else {
-            // Safe "unchecked" cast because newTimelock <= MAX_TIMELOCK.
-            pendingTimelock = PendingUint192(uint192(newTimelock), uint64(block.timestamp));
+            pendingTimelock.update(
+                // Safe "unchecked" cast because newTimelock <= MAX_TIMELOCK.
+                uint192(newTimelock),
+                timelock
+            );
```

Comment:
```suggestion
            // Safe "unchecked" cast because newTimelock <= MAX_TIMELOCK.
            pendingTimelock.update(uint192(newTimelock), timelock);
```
just to make it more readable (personal opinion)